### ISSUE TITLE
[Question] Is login_oauth_simple still relevant?

### ISSUE THREAD

**Issue by GioF71** [tamland/python-tidal#284]
Hello, in an effort of cleanup up the code of my tidal plugin for upmpdcli and remove unnecessary old stuff, I was wondering whether the login_oauth_simple method is still relevant. It seems to me that it eventually calls the login_session_file method, but I am not sure if something is different from calling that method just after creating an instance of the Session class.

OTOH, is there still a way of forcing streaming from FLAC files and avoiding MPEG-DASH, even if this means to lose access to hires content? That would still leave the change to select players that are not able to natively process MPEG-DASH.
If that is not possible, I'd like to clarify that in the documentation of the plugin.

Thank you for any help!

**Comment by tehkillerbee** [comment]
> whether the login_oauth_simple method is still relevant ... It seems to me that it eventually calls the login_session_file
Well currently, yes. login_session_file() calls login_oauth_simple() (or login_pkce().) so it is definitely used. You have multiple ways to login, but I suggest using the login_session_file() in most cases.

I do think we should cleanup the methods related to login, perhaps combine them and make some of them protected member classes to avoid calling them directly. 

> OTOH, is there still a way of forcing streaming from FLAC files and avoiding MPEG-DASH

Yes, by setting the quality to high_lossless, you can still use the get_url() method.

**Comment by GioF71** [comment]
Hello, getting back to this, I have to tell you that I don't use the login_oauth_simple() anymore in my code. Would you accept a pull request removing this method and cleaning up the code if needed
?

**Comment by tehkillerbee** [comment]
Well it is used in python-tidal, so we cannot remove the method (yet). See my previous answer.

> ... login_session_file() calls login_oauth_simple() (or login_pkce().) so it is definitely used. You have multiple ways to login, but I suggest using the login_session_file() in most cases.

**Comment by GioF71** [comment]
> Well it is used in python-tidal, so we cannot remove the method (yet). See my previous answer.
> 
> > ... login_session_file() calls login_oauth_simple() (or login_pkce().) so it is definitely used. You have multiple ways to login, but I suggest using the login_session_file() in most cases.

yes, sure you are right, maybe we can make it private as you were saying?
